public/static-checkout/js/scripts.js

```diff
@@ -0,0 +1,313 @@
+var stateSelect = document.getElementById('state');
+var countryStates = {
+  CA: {
+    AB: 'Alberta',
+    BC: 'British Columbia',
+    MB: 'Manitoba',
+    NB: 'New Brunswick',
+    NL: 'Newfoundland and Labrador',
+    NS: 'Nova Scotia',
+    NT: 'Northwest Territories',
+    NU: 'Nunavut',
+    ON: 'Ontario',
+    PE: 'Prince Edward Island',
+    QC: 'Québec',
+    SK: 'Saskatchewan',
+    YT: 'Yukon',
+  },
+  US: {
+    AL: 'Alabama',
+    AK: 'Alaska',
+    AZ: 'Arizona',
+    AR: 'Arkansas',
+    AA: 'Armed Forces Americas',
+    AE: 'Armed Forces Europe',
+    AP: 'Armed Forces Pacific',
+    CA: 'California',
+    CO: 'Colorado',
+    CT: 'Connecticut',
+    DE: 'Delaware',
+    DC: 'District Of Columbia',
+    FL: 'Florida',
+    GA: 'Georgia',
+    HI: 'Hawaii',
+    ID: 'Idaho',
+    IL: 'Illinois',
+    IN: 'Indiana',
+    IA: 'Iowa',
+    KS: 'Kansas',
+    KY: 'Kentucky',
+    LA: 'Louisiana',
+    ME: 'Maine',
+    MD: 'Maryland',
+    MA: 'Massachusetts',
+    MI: 'Michigan',
+    MN: 'Minnesota',
+    MS: 'Mississippi',
+    MO: 'Missouri',
+    MT: 'Montana',
+    NE: 'Nebraska',
+    NV: 'Nevada',
+    NH: 'New Hampshire',
+    NJ: 'New Jersey',
+    NM: 'New Mexico',
+    NY: 'New York',
+    NC: 'North Carolina',
+    ND: 'North Dakota',
+    OH: 'Ohio',
+    OK: 'Oklahoma',
+    OR: 'Oregon',
+    PA: 'Pennsylvania',
+    RI: 'Rhode Island',
+    SC: 'South Carolina',
+    SD: 'South Dakota',
+    TN: 'Tennessee',
+    TX: 'Texas',
+    UT: 'Utah',
+    VT: 'Vermont',
+    VA: 'Virginia',
+    WA: 'Washington',
+    WV: 'West Virginia',
+    WI: 'Wisconsin',
+    WY: 'Wyoming',
+  },
+};
+var ecommerceBaseUrl;
+if (window.location.hostname === 'payment.edx.org') {
+  ecommerceBaseUrl = 'https://ecommerce.edx.org';
+} else if (window.location.hostname === 'payment.stage.edx.org') {
+  ecommerceBaseUrl = 'https://ecommerce.stage.edx.org';
+} else if (window.location.hostname === 'localhost') {
+  ecommerceBaseUrl = 'http://localhost:18130';
+}
+
+userButton.addEventListener('click', function () {
+  document.querySelector('.dropdown-menu').classList.toggle('show');
+}, false);
+
+window.onclick = function closeMenu(event) {
+  if (!event.target.matches('.user-button')) {
+    var dropdownMenu = document.getElementsByClassName('dropdown-menu');
+    for (var i = 0; i < dropdownMenu.length; i++) {
+      var open = dropdownMenu[i];
+      if (open.classList.contains('show')) {
+        open.classList.remove('show');
+      }
+    }
+  }
+};
+
+function redirectToMFE(couponCode) {
+  var url = window.location.href;
+  var sku = /static-checkout\/([^&#]*).html/.exec(url)[1];
+  var newUrl = ecommerceBaseUrl + '/basket/add/?sku=' + sku;
+  if (couponCode) {
+    newUrl = newUrl.concat('&coupon=', couponCode);
+  }
+  window.location.href = newUrl;
+}
+
+function getBasketId() {
+  var url = window.location.href;
+  var basketParams = /[?&]basketId=([^&#]*)/.exec(url);
+  if (!basketParams) {
```

Comment:
do we expect this to happen or is this just defensive coding?

Reply:
Just added for in case there is so basket id in the url - defensive coding